### ISSUE TITLE
Implement tenant support

### ISSUE THREAD

**Issue by previ** [gravitee-io/issues#2275]
Implement tenant support.

### Expected Behavior
It should be possible to define context paths ( /service-a/, /service-b/ ) separately managed by groups of users.
A group should able to publish APIs on the tentant(s) whose it has admin role, and not be able to administer others APIs in other tenants.

### Current Behavior
Tenants are not implemented: users with admin role can manage all APIs.

**Comment by brasseld** [comment]
Correct me if I'm wrong but as far as I understand, you're looking to organize the management of the APIs between multiple groups. Right ?

**Comment by previ** [comment]
Yes, but the point is to avoid one group (team) be able to publish an API in the context path of another team's service or project.

**Comment by brasseld** [comment]
Ok, so it's not only about management.

So, IMHO, the best option would be to define context-path constraints per group.
For example, a constraint for a given group would be: the context path of the API must start with /groupa.

What do you think ?

**Comment by previ** [comment]
Exactly: there should be a new entity, say "Tenant" or "API group" with an associated base path, like, say /service-1/ (or /staging/service-1/, /prod/service-1/) and should be possible to link administration roles for the "Tenant" to (one or more) group.

**Comment by brasseld** [comment]
Hi @previ 

FYI, this will be part of the Gravitee Cockpit.

Cockpit will provide multi-tenancy features to let administrators to manage organizations and environments. Each environment will have its own path constraints (based on the domain name).

**Comment by previ** [comment]
That is very interensting, I look forward to test the new feature.
